client/dev-tools-promo/index.tsx

```diff
@@ -0,0 +1,30 @@
+import page, { Context as PageJSContext } from '@automattic/calypso-router';
+import { makeLayout, render as clientRender } from 'calypso/controller';
+import { navigation, siteSelection, sites } from 'calypso/my-sites/controller';
+import { siteDashboard } from 'calypso/sites-dashboard-v2/controller';
+import { DOTCOM_DEVELOPER_TOOLS_PROMO } from 'calypso/sites-dashboard-v2/site-preview-pane/constants';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { devToolsPromo } from './controller';
+
+const redirectForNonSimpleSite = ( context: PageJSContext, next: () => void ) => {
+	const state = context.store.getState();
+	const site = getSelectedSite( state );
+	if ( site && site.jetpack ) {
+		return page.redirect( `/hosting/${ context.params.site }` );
```

Comment:
I saw that the redirect was done differently in `redirectHomeIfIneligible`, they use `context.page.replace` to redirect. One cool side effect of page.redirect is that it seems to "remember" that you were supposed to be on the dev tools tab.

e.g. if you're on the dev tools tab for SITE_A, then you change to an atomic site SITE_B, it would redirect you to the /hosting/SITE_B page. then if you go to another simple site SITE_C, it would take you back to the dev tools tab for that site `/dev-tools-promo/SITE_C`.

The context.page.replace approach would have taken you to `/hosting/SITE_C`